readthedocs/doc_builder/environments.py

```diff
@@ -444,24 +444,21 @@ def run_command_class(
             self.commands.append(build_cmd)
 
         if build_cmd.failed:
-            msg = 'Command {cmd} failed'.format(cmd=build_cmd.get_command())
-
-            # TODO: improve this error report. This is showing _the full_
-            # stdout to the user exposing it at the top of the Build Detail's
-            # page in red. It would be good to reduce the noise here and just
-            # point the user to take a look at its output from the command's
-            # output itself.
-            if build_cmd.output:
-                msg += ':\n{out}'.format(out=build_cmd.output)
-
             if warn_only:
+                msg = 'Command {cmd} failed'.format(cmd=build_cmd.get_command())
+                if build_cmd.output:
+                    msg += ':\n{out}'.format(out=build_cmd.output)
                 log.warning(
                     msg,
                     project_slug=self.project.slug if self.project else '',
                     version_slug=self.version.slug if self.version else '',
                 )
             else:
-                raise BuildUserError(msg)
+                # TODO: for now, this still outputs a generic error message
+                # that is the same across all commands. We could improve this
+                # with more granular error messages that vary by the command
+                # being run.
+                raise BuildUserError()
```

Comment:
I remembered that you noticed this in the original PR at https://github.com/readthedocs/readthedocs.org/pull/8815#discussion_r791120625

What about passing this instead to show a better message in this case? Still generic, but pointing to a problem in the project instead of the app
```suggestion
                raise BuildUserError(BuildUserError.GENERIC)
```

Reply:
I should have noted, this is indeed the error I'm trying to raise. If you think explicit is better here, :+1: . I was keeping this logic centralized inside `on_failure`, letting this logic handle the missing `msg`:

https://github.com/readthedocs/readthedocs.org/blob/26e371b68f9369a7e87a1d2885d6d357a018e7fc/readthedocs/projects/tasks/builds.py#L380-L384

Reply:
Oh, yeah! You are right 👍

I did a quick review while deploying, but your code is perfect 😎